rpcs3/Emu/System.cpp

```diff
@@ -96,11 +158,76 @@ void fmt_class_string<game_boot_result>::format(std::string& out, u64 arg)
 		case game_boot_result::file_creation_error: return "Could not create important files";
 		case game_boot_result::firmware_missing: return "Firmware is missing";
 		case game_boot_result::unsupported_disc_type: return "This disc type is not supported yet";
+		case game_boot_result::savestate_corrupted: return "Savestate data is corrupted or it's not an RPCS3 savestate";
+		case game_boot_result::savestate_version_unsupported: return "Savestate versioning data differes from your RPCS3 build";
 		}
 		return unknown;
 	});
 }
 
+namespace rsx
+{
+	struct avconf;
+}
+
+class np_handler;
+struct loaded_npdrm_keys;
+struct lv2_socket;
+struct lv2_memory_container;
+struct lv2_spu_group;
+struct sys_vm_t;
+struct pad_info;
+struct sce_np_trophy_manager;
+struct sysutil_cb_manager;
+struct gem_config;
+struct music_state;
+class cell_audio_thread;
+class usb_handler_thread;
+class camera_context;
+
+// This function ensures constant initialization order between different compilers and builds 
+void init_fxo_for_exec(utils::serial* ar, bool full = false)
```

Comment:
I don't like that this file is getting bloated again.
Maybe try to put that function in its own file or somewhere else